controllers/admin/AdminImagesController.php

```diff
@@ -680,10 +686,15 @@ protected function _regenerateNewImages($dir, $type, $productsImages = false)
 
         /*
          * Let's resolve which formats we will use for image generation.
+         * In new image system, it's multiple formats. In case of legacy, it's only .jpg.
          *
          * In case of .jpg images, the actual format inside is decided by ImageManager.
          */
-        $configuredImageFormats = $this->imageFormatConfiguration->getGenerationFormats();
+        if ($this->isMultipleImageFormatFeatureEnabled) {
+            $configuredImageFormats = $this->imageFormatConfiguration->getGenerationFormats();
+        } else {
+            $configuredImageFormats = ['jpg'];
+        }
```

Comment:
This check is not supposed to be needed `ImageFormatConfiguration` already handles it internally and returns `['jpg']` automatically when the feature flag is disabled, that's what you handled in your previous PR

I think just the modif in `beforeUpdateOptions` should be enough no?

Reply:
```suggestion
        $configuredImageFormats = $this->imageFormatConfiguration->getGenerationFormats();
```